packages/node-resolve/src/index.js

```diff
@@ -19,15 +17,15 @@ import {
 const builtins = new Set(builtinList);
 const ES6_BROWSER_EMPTY = '\0node-resolve:empty.js';
 const nullFn = () => null;
-const defaults = {
-  customResolveOptions: {},
-  dedupe: [],
+const defaults = Object.freeze({
```

Comment:
I'd rather see a simple convenience method (`const deepFreeze = () =>`) in here to take the place of the removed dep on line 30, rather than a buncha calls to `Object.freeze` that we'd have to remember to add to anything new to `defaults`. A depth of 4 seems like it would be sufficient. The code _is_ [rather simple](https://github.com/substack/deep-freeze/blob/master/index.js).